### PR TITLE
feat(trie): use branch node hash masks in sparse trie

### PR DESCRIPTION
**Pull request by shekhirin**
Integrates https://github.com/paradigmxyz/reth/pull/13129 into the sparse trie, revealing the hash masks of branch nodes along with the proofs.

### CODE REVIEW

File: crates/trie/sparse/src/trie.rs [hunk 1]
```diff
@@ -933,7 +976,8 @@ where
                             if let Some(node) = self.provider.blinded_node(child_path.clone())? {
                                 let decoded = TrieNode::decode(&mut &node[..])?;
                                 trace!(target: "trie::sparse", ?child_path, ?decoded, "Revealing remaining blinded branch child");
-                                self.reveal_node(child_path.clone(), decoded)?;
+                                // TODO: fetch hash mask
```
Comment (rkrasiuk) [inline crates/trie/sparse/src/trie.rs:979]: todo?

File: crates/trie/sparse/src/state.rs [hunk 2]
```diff
@@ -102,75 +102,6 @@ impl<F: BlindedProviderFactory> SparseStateTrie<F> {
         self.storages.get_mut(account).and_then(|e| e.as_revealed_mut())
     }
 
-    /// Reveal unknown trie paths from provided leaf path and its proof for the account.
-    /// NOTE: This method does not extensively validate the proof.
-    pub fn reveal_account(
```
Comment (rkrasiuk) [inline crates/trie/sparse/src/state.rs:107]: i'd prefer to keep this methods, they are still useful. mb with `debug_assert!(!self.updates_enabled)`

File: crates/trie/sparse/src/trie.rs (hunk 1 above)
Comment (shekhirin) [reply crates/trie/sparse/src/trie.rs:979]: plan to address it in the next PR, as it requires modifying the `BlindedProvider` trait

File: crates/trie/sparse/src/state.rs (hunk 2 above)
Comment (shekhirin) [reply crates/trie/sparse/src/state.rs:107]: fair, useful at least for tests

File: crates/trie/sparse/src/trie.rs (hunk 1 above)
Comment (shekhirin) [reply crates/trie/sparse/src/trie.rs]: replaced with a comment

**Review by rkrasiuk — APPROVED** [review]

File: crates/trie/sparse/src/state.rs [hunk 3]
```diff
@@ -186,34 +198,48 @@ impl<F: BlindedProviderFactory> SparseStateTrie<F> {
...
                     self.retain_updates,
                 )?;
 
                 // Reveal the remaining proof nodes.
-                for (path, bytes) in storage_nodes {
+                for (path, bytes) in nodes {
                     let node = TrieNode::decode(&mut &bytes[..])?;
-                    trace!(target: "trie::sparse", ?account, ?path, ?node, "Revealing storage node");
-                    trie.reveal_node(path, node)?;
+                    let hash_mask = if let TrieNode::Branch(_) = node {
```
Comment (mattsse) [inline crates/trie/sparse/src/state.rs:235]: looks like we could use `fn is_branch()`

suggestion for followup

**Review by mattsse — APPROVED** [review]

**Pull request merged**